### ISSUE TITLE
Code refactoring for e2e tests

### ISSUE THREAD

**Issue by ashahi1** [vmware/vsphere-storage-for-docker#1470]
- [ ] Create three separate test files for vmgroup tests - one for default vmgroup tests,  another for user vmgroup tests and one for miscellaneous tests related to vmgroups
https://github.com/vmware/docker-volume-vsphere/pull/1381#pullrequestreview-45707021

- [ ] Create a single suite for all tests that are doing config init/remove so we initialize and remove the DB once at the test-suite level rather than each test doing it.

**Comment by shaominchen** [comment]
Moving refactoring tasks to P2.

**Comment by ashahi1** [comment]
- [ ] Create a single suite for all tests that are doing config init/remove so we initialize and remove the DB once at the test-suite level rather than each test doing it.

For above task, vmgroup tests are spread across four different files:
- basic_test.go
- defaultvmgroup_test.go
- vmgroupmisc_test.go
- vmgroups_test.go

I am thinking of following solution which involves having a single VMgroupTestSuite struct
1. Creating single SetUpSuite and TearDownSuite. 
2. Same SetUpSuite and TearDownSuite will be shared among all vmgroup tests spread across four separate files.
3. Config init will be done only once in SetUpSuite and then run all the vmgroup tests.
4. Config remove will be done only once in TearDownSuite once all vmgroup tests are finished.

So we need to have common SetUpSuite and TearDownSuite for all the vmgroup tests.

Now, the tests under vmgroups_test.go are doing lots of stuff in their SetUpSuite - creating vmgroups and adding VMs to each vmgroup. 
Similarly, in their TearDownSuite, corresponding removal of VM from vmgroup and deletion of vmgroup takes place.

So basically, we will have to make considerable code changes or we might have to re-write whole test.

Hence, before I proceed with this task, just wanted to check if I should use this approach so as to save on going back and forth during reviews and do we really need to make this change now.

/cc @govint @shaominchen @shuklanirdesh82 @pshahzeb @lipingxue

**Comment by shaominchen** [comment]
It makes sense to have separate files (or test suites) for default vmgroup tests and user-defined vmgroup tests. I'm not sure if we really need a separate file for miscellaneous tests related to vmgroups - what are those exactly? A vmgroup should be either default or non-default.

As for "Config init/remove should be done only once", I don't think this is really critical or even necessary since our DB is just a local file. It's actually pretty lightweight (comparing to other expensive costs that we spent in e2e tests) to do config init/remove. So personally I feel this is low priority and we may just keep it as is.

**Comment by shuklanirdesh82** [comment]
not a high priority
